src/scib_metrics/utils/_lisi.py

```diff
@@ -0,0 +1,115 @@
+from typing import Tuple, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+from flax import struct
+
+NdArray = Union[np.ndarray, jnp.ndarray]
+
+
+@struct.dataclass
```

Comment:
https://github.com/deepmind/chex#dataclass-dataclasspy

Let's actually use this as it's better to depend on chex for this project, as we can use it in the test suite as well